src/app/shared/services/profile.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http'
+import { Observable } from 'rxjs';
+import { User } from 'src/app/models/user.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProfileService {
+  public baseUrl: string;
+  public apiPath: string;
+  constructor(private http: HttpClient) { }
+
+
+  /**
+   * Returns user object
+   */
+  public getProfileData(): Observable<User> {
+    return Observable.create((observer: any) => {
```

Comment:
why we have set timeout explicitly?

Reply:
For the time being I'm using Mock endpoints. In the future when the API is available we can integrate together with it.